### PR TITLE
Remove newline characters from fields before generating NACHA files

### PR DESCRIPTION
**Pull request by zcotter**

### CODE REVIEW

File: lib/ach_client/providers/sftp/ach_transaction.rb [hunk 1]
```diff
@@ -13,20 +13,25 @@ def do_send
...
           self.class.parent.originating_dfi_identification
+        )
         entry.trace_number = external_ach_id.gsub(/\D/, '').to_i # Does need to be a number
         entry
       end
 
       private
 
+      def remove_newlines(value)
+        value.gsub(/[\n\r]/, '')
```
Comment (octopushugs) [inline lib/ach_client/providers/sftp/ach_transaction.rb]: @zcotter This method pretty much self-documenting, but would you mind adding a comment explaining the intended behavior of the regex?

File: lib/ach_client/providers/sftp/ach_transaction.rb [hunk 2]
```diff
@@ -13,20 +13,25 @@ def do_send
...
         entry.amount = amount_in_cents
-        entry.individual_id_number = external_ach_id # Doesn't need to be a number
-        entry.individual_name = merchant_name
-        entry.originating_dfi_identification =
+        entry.individual_id_number = remove_newlines(external_ach_id) # Doesn't need to be a number
+        entry.individual_name = remove_newlines(merchant_name)
+        entry.originating_dfi_identification = remove_newlines(
           self.class.parent.originating_dfi_identification
+        )
         entry.trace_number = external_ach_id.gsub(/\D/, '').to_i # Does need to be a number
```
Comment (natseg) [inline lib/ach_client/providers/sftp/ach_transaction.rb]: I would also extract this similarly to `remove_newlines` to be very intentional about normalizing the data.

Comment (zcotter) [reply lib/ach_client/providers/sftp/ach_transaction.rb]: Yeah I agree, not clear what this does so I'll make it a method

File: lib/ach_client/providers/sftp/ach_transaction.rb (hunk 1 above)
Comment (nixpulvis) [reply lib/ach_client/providers/sftp/ach_transaction.rb]: Heh, always comment regexes, I agree.

**Pull request merged**